src/main/java/hudson/plugins/git/extensions/impl/SubmoduleOption.java

```diff
@@ -74,7 +94,30 @@ public void onCheckoutCompleted(GitSCM scm, AbstractBuild<?, ?> build, GitClient
             // This ensures we don't miss changes to submodule paths and allows
             // seamless use of bare and non-bare superproject repositories.
             git.setupSubmoduleUrls(revToBuild.lastBuild.getRevision(), listener);
-            git.submoduleUpdate().recursive(recursiveSubmodules).remoteTracking(trackingSubmodules).execute();
+
+            SubmoduleUpdateCommand upd = git.submoduleUpdate().recursive(recursiveSubmodules).remoteTracking(trackingSubmodules);
+            if (submoduleBranches != null) {
+                EnvVars env = build.getEnvironment();
+                for (SubmoduleBranch sb : submoduleBranches) {
+                    String expandedBranchValue = env.expand(sb.getBranch());
+
+                    //
+                    //  If nothing gets replaced, we are probably in the initial checkout of the
```

Comment:
I am not sure if I understand what you are talking about here. Can you explain it on a dedicated example?